### PR TITLE
feat(list): add nav list styles

### PR DESCRIPTION
**Pull request by kara**
Basic PR for nav list styles.

Still TODO:
- Discuss approach for shared focus strategy and add
- Add sugared syntax for basic nav lists

### CODE REVIEW

File: src/core/style/_palette.scss [hunk 1]
```diff
@@ -342,6 +343,7 @@ $md-dark-theme-background: (
   status-bar: black,
   app-bar:    map_get($md-grey, 900),
   background: #303030,
+  hover:             rgba(white, 0.04),
```
Comment (jelbourn) [inline src/core/style/_palette.scss]: Can we add a TODO for the `hover` state to consult with Material Design UX?

File: src/demo-app/demo-app.html [hunk 2]
```diff
@@ -1,25 +1,34 @@
...
-  </ul>
-  <button md-raised-button (click)="root.dir = (root.dir == 'rtl' ? 'ltr' : 'rtl')">
-    {{root.dir.toUpperCase()}}
-  </button>
-  <div #root="$implicit" dir="ltr" class="demo-content">
-    <router-outlet></router-outlet>
-  </div>
-</div>
+<md-sidenav-layout>
+    <md-sidenav #start>
```
Comment (princemaple) [inline src/demo-app/demo-app.html]: indentation seems off

Comment (kara) [inline src/demo-app/demo-app.html]: The indentation was indeed off; thanks for noticing!

**Comment by jelbourn** [comment]
LGTM

**Pull request merged**